src/server/routes/sockets.js

```diff
@@ -5,6 +5,12 @@ const qs = require('querystring');
 const io = require('./../io').get();
 
 io.on('connection', function (socket) {
+  let referer = socket.handshake.headers.referer;
```

Comment:
Can you elaborate on what this chunk of code is doing?

Reply:
@gaelder Can handle this one. He wrote the backend parts of this PR.

Reply:
@d2fong This grabs the url that the client is currently viewing. It is used to determine which pathway is being viewed.